cmd/service_linux.go

```diff
@@ -0,0 +1,316 @@
+	Example:           "cloudfuse service uninstall",
+	FlagErrorHandling: cobra.ExitOnError,
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		// 1. find and remove cloudfuse.service from /etc/systemd/system and run systemctl daemon-reload
+		removeFileCmd := exec.Command("sudo", "rm", "/etc/systemd/system/cloudfuse.service")
+		err := removeFileCmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to delete cloudfuse.service file from /etc/systemd/system due to following error: [%s]", err.Error())
+		}
+
+		systemctlDaemonReloadCmd := exec.Command("sudo", "systemctl", "daemon-reload")
+		err = systemctlDaemonReloadCmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to run 'systemctl daemon-reload' command due to following error: [%s]", err.Error())
+		}
+
+		return nil
+	},
+}
+
+//--------------- command section ends
+
+func collectServiceData(serviceFilePath string) (map[string]string, error) {
+	serviceFile, err := os.Open("./setup/cloudfuse.service")
+
+	if err != nil {
+		fmt.Println("Error opening file:", err)
+		return nil, err
+	}
+
+	defer serviceFile.Close()
+
+	scanner := bufio.NewScanner(serviceFile)
+	serviceData := make(map[string]string)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.Contains(line, "User=") {
+			parts := strings.SplitN(line, "=", 2)
+			key := strings.TrimSpace(parts[0])
+			value := strings.TrimSpace(parts[1])
+			serviceData[key] = value
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		fmt.Println("Error reading file:", err)
+		return nil, err
+	}
+	return serviceData, nil
+}
+
+func newSericeFile(mountPath string, configPath string, dir string) (string, error) {
+
+	//mountpath or config?
+	var oldString string
+	var newString string
+
+	oldConfigStr := "Environment=ConfigFile=/path/to/config/file/config.yaml"
+	newConfigStr := fmt.Sprintf("Environment=ConfigFile=%s", common.JoinUnixFilepath(dir, configPath))
+
+	oldMountStr := "Environment=MoutingPoint=/path/to/mounting/point"
+	newMountStr := fmt.Sprintf("Environment=MoutingPoint=%s", common.JoinUnixFilepath(dir, mountPath))
+
+	// open service defaultFile for read write
+	defaultFile, err := os.OpenFile("./setup/cloudfuse.service", os.O_RDWR, 0644)
+	if err != nil {
+		return "", fmt.Errorf("error opening file: [%s]", err.Error())
+	}
+	defer defaultFile.Close()
+
+	scanner := bufio.NewScanner(defaultFile)
+	var lines []string
+
+	// Read the file line by line
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		// Check if the line contains the search string
+		if strings.Contains(line, "MoutingPoint") {
+			// Modify the line by replacing the old string with the new string
+			line = strings.ReplaceAll(line, oldMountStr, newMountStr)
+		}
+		if strings.Contains(line, "ConfigFile") {
+			line = strings.ReplaceAll(line, oldConfigStr, newConfigStr)
+		}
+
+		// add the -o default_permissions if not present
+		if strings.Contains(line, "ExecStart") && !strings.Contains(line, "-o allow_other") {
```

Comment:
Why are you checking for this if we write the config file. Shouldn't the written config file already have allow_other?

Reply:
There is no guarantee that the config file will have this option. It's easier to supply this option here instead of checking for it in the config file.

Reply:
I guess what I mean is if we are writing the systemd file ourselves, then we don't need to read it and then determine if we should add it if we are always adding that to the systemd file.